content/courses/ada-idioms/chapters/abstract_data_types.rst

```diff
@@ -331,13 +338,14 @@ change in client usage, such as performance changes, but it will not be a
 matter of the legality of the client code. Illegal client usage of an ADT
 wouldn't have compiled successfully in the first place.
 
-The private type is the fundamental approach to creating abstractions in Ada,
-just as the use of the *public*, *private*, and *protected* parts of classes is
-fundamental to creating abstractions in class-oriented languages. Not every
-type can be private, as illustrated by the client expectation for array
-indexing in Ada. Not every type should be private, for example those that are
-explicitly numeric. But the ADT should be the default design idiom when
-decomposing a problem into a solution.
+The private type is the fundamental approach to creating abstractions in
+Ada, just as the use of the *public*, *private*, and *protected* parts of
+classes is fundamental to creating abstractions in class-oriented
+languages. Not every type can be private, as illustrated by the client
+expectation for array indexing in Ada prior to Ada 2012. Not every type
```

Comment:
Adding "prior to Ada 2012" here makes the uninformed reader wonder why is it now different in Ada 2012... Maybe we should come up with better examples of the necessity of non-private types.